### ISSUE TITLE
Large Blob Download Fails with "Received Response with Content" or "ContentDecodingError"

### ISSUE THREAD

**Issue by brian-l-coker** [Azure/azure-cli#4490]
### Description
We recently upgraded to 2.0.16 & 2.0.17 and noticed that download of large blobs fail. We've tried the following test and found that large blobs **uploaded using 2.0.16 and 2.0.17** cannot be downloaded with CLI 2.0.16+. Large blobs **uploaded using earlier versions download just fine**.

If you upload a large blob (e.g. tar.gz > 43 MB) with 2.0.16, you will receive a message during download of: "Received Response with Content"

If you upload a large blob (e.g. tar.gz > 43 MB) with 2.0.17, you will receive a message during download of: "ContentDecodingError"

Here is the upload log using 2.0.17:
```
[bcoker bin]$ ./az storage blob upload --container-name test --file ~/git/sample_serviceB/target/deploy/serviceBApplication.tar.gz --name serviceBApplication_with_2.0.17.tar.gz --account-name XXXX --account-key XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --debug
File logging enabled - Writing logs to '/home/bcoker/.azure/logs'.
Command arguments ['storage', 'blob', 'upload', '--container-name', 'test', '--file', '/home/bcoker/git/sample_serviceB/target/deploy/serviceBApplication.tar.gz', '--name', 'serviceBApplication_with_2.0.17.tar.gz', '--account-name', 'XXXX', '--account-key', 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX']
Current cloud config:
{'endpoints': {'active_directory': 'https://login.microsoftonline.com',
               'active_directory_data_lake_resource_id': 'https://datalake.azure.net/',
               'active_directory_graph_resource_id': 'https://graph.windows.net/',
               'active_directory_resource_id': 'https://management.core.windows.net/',
               'batch_resource_id': 'https://batch.core.windows.net/',
               'gallery': 'https://gallery.azure.com/',
               'management': 'https://management.core.windows.net/',
               'resource_manager': 'https://management.azure.com/',
               'sql_management': 'https://management.core.windows.net:8443/',
               'vm_image_alias_doc': 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json'},
 'is_active': True,
 'name': 'AzureCloud',
 'profile': 'latest',
 'suffixes': {'azure_datalake_analytics_catalog_and_job_endpoint': 'azuredatalakeanalytics.net',
              'azure_datalake_store_file_system_endpoint': 'azuredatalakestore.net',
              'keyvault_dns': '.vault.azure.net',
              'sql_server_hostname': '.database.windows.net',
              'storage_endpoint': 'core.windows.net'}}
Registered application event handler 'CommandTableParams.Loaded' at <function add_id_parameters at 0x195d848>
Registered application event handler 'CommandTable.Loaded' at <function add_id_parameters at 0x195d848>
Successfully loaded command table from module 'storage'.
Extensions directory: '/home/bcoker/.azure/cliextensions'
Application event 'CommandTable.Loaded' with event data {'command_table': {'storage blob upload': <azure.cli.core.commands.CliCommand object at 0x1981950>}}
Application event 'CommandParser.Loaded' with event data {'parser': AzCliCommandParser(prog='az', usage=None, description=None, version=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)}
Application event 'CommandTableParams.Loaded' with event data {'command_table': {'storage blob upload': <azure.cli.core.commands.CliCommand object at 0x1981950>}}
Application event 'CommandParser.Parsing' with event data {'argv': ['storage', 'blob', 'upload', '--container-name', 'test', '--file', '/home/bcoker/git/sample_serviceB/target/deploy/serviceBApplication.tar.gz', '--name', 'serviceBApplication_with_2.0.17.tar.gz', '--account-name', 'XXXX', '--account-key', 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX']}
Application event 'CommandParser.Parsed' with event data {'args': Namespace(_command_package='storage', _jmespath_query=None, _log_verbosity_debug=False, _log_verbosity_verbose=False, _output_format='jsonc', _parser=AzCliCommandParser(prog='az storage blob upload', usage=None, description='Creates a new blob from a file path, or updates the content of an existing blob, with automatic chunking and progress notifications.', version=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True), _subscription_id=No [...]
Getting data service client service_type=BlockBlobService
urllib3.connectionpool : Starting new HTTPS connection (1): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "PUT /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 201 None
Finished[#############################################################]  100.0000%
Application event 'Application.TransformResults' with event data {'event_data': {'result': {'lastModified': '2017-09-19T21:03:58+00:00', 'etag': '"0x8D4FFA1F247998C"'}}}
Application event 'Application.FilterResults' with event data {'event_data': {'result': {'lastModified': '2017-09-19T21:03:58+00:00', 'etag': '"0x8D4FFA1F247998C"'}}}
{
  "etag": "\"0x8D4FFA1F247998C\"",
  "lastModified": "2017-09-19T21:03:58+00:00"
}
```

And the download log using 2.0.17:
```
[bcoker bin]$ ./az storage blob download --container-name test --file /tmp/serviceBApplication.tar.gz --name serviceBApplication_with_2.0.17.tar.gz --account-name XXXX --account-key XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --debug
File logging enabled - Writing logs to '/home/bcoker/.azure/logs'.
Command arguments ['storage', 'blob', 'download', '--container-name', 'test', '--file', '/tmp/serviceBApplication.tar.gz', '--name', 'serviceBApplication_with_2.0.17.tar.gz', '--account-name', 'XXXX', '--account-key', 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX']
Current cloud config:
{'endpoints': {'active_directory': 'https://login.microsoftonline.com',
               'active_directory_data_lake_resource_id': 'https://datalake.azure.net/',
               'active_directory_graph_resource_id': 'https://graph.windows.net/',
               'active_directory_resource_id': 'https://management.core.windows.net/',
               'batch_resource_id': 'https://batch.core.windows.net/',
               'gallery': 'https://gallery.azure.com/',
               'management': 'https://management.core.windows.net/',
               'resource_manager': 'https://management.azure.com/',
               'sql_management': 'https://management.core.windows.net:8443/',
               'vm_image_alias_doc': 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json'},
 'is_active': True,
 'name': 'AzureCloud',
 'profile': 'latest',
 'suffixes': {'azure_datalake_analytics_catalog_and_job_endpoint': 'azuredatalakeanalytics.net',
              'azure_datalake_store_file_system_endpoint': 'azuredatalakestore.net',
              'keyvault_dns': '.vault.azure.net',
              'sql_server_hostname': '.database.windows.net',
              'storage_endpoint': 'core.windows.net'}}
Registered application event handler 'CommandTableParams.Loaded' at <function add_id_parameters at 0x1433848>
Registered application event handler 'CommandTable.Loaded' at <function add_id_parameters at 0x1433848>
Successfully loaded command table from module 'storage'.
Extensions directory: '/home/bcoker/.azure/cliextensions'
Application event 'CommandTable.Loaded' with event data {'command_table': {'storage blob download': <azure.cli.core.commands.CliCommand object at 0x14575d0>}}
Application event 'CommandParser.Loaded' with event data {'parser': AzCliCommandParser(prog='az', usage=None, description=None, version=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)}
Application event 'CommandTableParams.Loaded' with event data {'command_table': {'storage blob download': <azure.cli.core.commands.CliCommand object at 0x14575d0>}}
Application event 'CommandParser.Parsing' with event data {'argv': ['storage', 'blob', 'download', '--container-name', 'test', '--file', '/tmp/serviceBApplication.tar.gz', '--name', 'serviceBApplication_with_2.0.17.tar.gz', '--account-name', 'XXXX', '--account-key', 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX']}
Application event 'CommandParser.Parsed' with event data {'args': Namespace(_command_package='storage', _jmespath_query=None, _log_verbosity_debug=False, _log_verbosity_verbose=False, _output_format='jsonc', _parser=AzCliCommandParser(prog='az storage blob download', usage=None, description='Downloads a blob to a file path, with automatic chunking and progress notifications. Returns an instance of :class:`Blob` with  properties and metadata.', version=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True [...]
Getting data service client service_type=BlockBlobService
urllib3.connectionpool : Starting new HTTPS connection (1): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 33554432
urllib3.connectionpool : Starting new HTTPS connection (2): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : Starting new HTTPS connection (3): XXXX.blob.core.windows.net
urllib3.connectionpool : Starting new HTTPS connection (4): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : Starting new HTTPS connection (5): XXXX.blob.core.windows.net
urllib3.connectionpool : Starting new HTTPS connection (6): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : Starting new HTTPS connection (7): XXXX.blob.core.windows.net
urllib3.connectionpool : Starting new HTTPS connection (8): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304
urllib3.connectionpool : Starting new HTTPS connection (9): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 4194304

ContentDecodingError
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 1277773
urllib3.connectionpool : Starting new HTTPS connection (10): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 1277773
urllib3.connectionpool : Starting new HTTPS connection (11): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 1277773
urllib3.connectionpool : Starting new HTTPS connection (12): XXXX.blob.core.windows.net
urllib3.connectionpool : https://XXXX.blob.core.windows.net:443 "GET /test/serviceBApplication_with_2.0.17.tar.gz HTTP/1.1" 206 1277773

```

Same problem appears to exist on the batch upload command as well. 

The problem has been replicated on multiple RHEL Virtual Machines as well as Windows 7 PCs. Older VM's that had version 2.0.7 and 2.0.14 work fine.

Blobs uploaded with 2.0.16 and 2.0.17 appear to download just fine with x-plat-cli, browser, and REST API.

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: MSI on Windows. Yum on RHEL per Instructions.

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0.16 & 2.0.17

**OS Version:** What OS and version are you using?  
Answer here: Windows 7 and RedHat 7.4

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: cmd.exe & bash

Python Modules for 2.0.17
```
adal (0.4.7)
applicationinsights (0.11.1)
argcomplete (1.9.2)
asn1crypto (0.22.0)
azure-batch (3.1.0)
azure-cli (2.0.17)
azure-cli-acr (2.0.11)
azure-cli-acs (2.0.15)
azure-cli-appservice (0.1.16)
azure-cli-batch (3.1.3)
azure-cli-billing (0.1.4)
azure-cli-cdn (0.0.8)
azure-cli-cloud (2.0.7)
azure-cli-cognitiveservices (0.1.7)
azure-cli-command-modules-nspkg (2.0.1)
azure-cli-component (2.0.7)
azure-cli-configure (2.0.10)
azure-cli-consumption (0.1.4)
azure-cli-container (0.1.10)
azure-cli-core (2.0.16)
azure-cli-cosmosdb (0.1.12)
azure-cli-dla (0.0.11)
azure-cli-dls (0.0.13)
azure-cli-eventgrid (0.1.3)
azure-cli-extension (0.0.2)
azure-cli-feedback (2.0.6)
azure-cli-find (0.2.6)
azure-cli-interactive (0.3.9)
azure-cli-iot (0.1.11)
azure-cli-keyvault (2.0.10)
azure-cli-lab (0.0.10)
azure-cli-monitor (0.0.9)
azure-cli-network (2.0.14)
azure-cli-nspkg (3.0.1)
azure-cli-profile (2.0.12)
azure-cli-rdbms (0.0.6)
azure-cli-redis (0.2.8)
azure-cli-resource (2.0.14)
azure-cli-role (2.0.11)
azure-cli-servicefabric (0.0.3)
azure-cli-sf (1.0.8)
azure-cli-sql (2.0.11)
azure-cli-storage (2.0.15)
azure-cli-vm (2.0.14)
azure-common (1.1.8)
azure-datalake-store (0.0.15)
azure-graphrbac (0.31.0)
azure-keyvault (0.3.4)
azure-mgmt-authorization (0.30.0)
azure-mgmt-batch (4.1.0)
azure-mgmt-billing (0.1.0)
azure-mgmt-cdn (0.30.2)
azure-mgmt-cognitiveservices (1.0.0)
azure-mgmt-compute (2.0.0)
azure-mgmt-consumption (0.1.0)
azure-mgmt-containerinstance (0.1.0)
azure-mgmt-containerregistry (0.3.1)
azure-mgmt-containerservice (1.0.0)
azure-mgmt-datalake-analytics (0.1.6)
azure-mgmt-datalake-nspkg (2.0.0)
azure-mgmt-datalake-store (0.1.6)
azure-mgmt-devtestlabs (2.0.0)
azure-mgmt-dns (1.0.1)
azure-mgmt-documentdb (0.1.3)
azure-mgmt-iothub (0.2.2)
azure-mgmt-keyvault (0.40.0)
azure-mgmt-monitor (0.2.1)
azure-mgmt-network (1.4.0)
azure-mgmt-nspkg (2.0.0)
azure-mgmt-rdbms (0.1.0)
azure-mgmt-redis (4.1.0)
azure-mgmt-resource (1.1.0)
azure-mgmt-servicefabric (0.1.0)
azure-mgmt-sql (0.8.0)
azure-mgmt-storage (1.2.0)
azure-mgmt-trafficmanager (0.30.0)
azure-mgmt-web (0.32.0)
azure-monitor (0.3.0)
azure-multiapi-storage (0.1.3)
azure-nspkg (2.0.0)
azure-servicefabric (5.6.130)
bcrypt (3.1.3)
certifi (2017.7.27.1)
cffi (1.10.0)
chardet (3.0.4)
colorama (0.3.9)
cryptography (2.0.3)
enum34 (1.1.6)
funcsigs (1.0.2)
futures (3.1.1)
gyp (0.1)
humanfriendly (4.4.1)
idna (2.6)
ipaddress (1.0.18)
isodate (0.5.4)
jmespath (0.9.3)
keyring (10.4.0)
mock (2.0.0)
monotonic (1.3)
msrest (0.4.14)
msrestazure (0.4.13)
ndg-httpsclient (0.4.3)
oauthlib (2.0.3)
paramiko (2.2.1)
pathlib2 (2.3.0)
pbr (3.1.1)
pip (9.0.1)
prompt-toolkit (1.0.15)
pyasn1 (0.3.4)
pycparser (2.18)
pydocumentdb (2.2.0)
Pygments (2.2.0)
PyJWT (1.5.3)
PyNaCl (1.1.2)
pyOpenSSL (17.2.0)
python-dateutil (2.6.1)
pytz (2017.2)
PyYAML (3.12)
requests (2.18.4)
requests-oauthlib (0.8.0)
scandir (1.5)
scp (0.10.2)
SecretStorage (2.3.1)
setuptools (36.4.0)
six (1.10.0)
sshtunnel (0.1.2)
tabulate (0.7.7)
urllib3 (1.22)
vsts-cd-manager (0.118.0)
wcwidth (0.1.7)
wheel (0.30.0)
Whoosh (2.7.4)
xmltodict (0.11.0)
```

Python Modules for 2.0.16
```
adal (0.4.7)
applicationinsights (0.11.0)
argcomplete (1.9.2)
asn1crypto (0.22.0)
azure-batch (3.1.0)
azure-cli (2.0.16)
azure-cli-acr (2.0.11)
azure-cli-acs (2.0.14)
azure-cli-appservice (0.1.15)
azure-cli-batch (3.1.2)
azure-cli-billing (0.1.4)
azure-cli-cdn (0.0.7)
azure-cli-cloud (2.0.7)
azure-cli-cognitiveservices (0.1.7)
azure-cli-command-modules-nspkg (2.0.1)
azure-cli-component (2.0.7)
azure-cli-configure (2.0.10)
azure-cli-consumption (0.1.4)
azure-cli-container (0.1.9)
azure-cli-core (2.0.15)
azure-cli-cosmosdb (0.1.12)
azure-cli-dla (0.0.11)
azure-cli-dls (0.0.13)
azure-cli-eventgrid (0.1.3)
azure-cli-feedback (2.0.6)
azure-cli-find (0.2.6)
azure-cli-interactive (0.3.9)
azure-cli-iot (0.1.11)
azure-cli-keyvault (2.0.9)
azure-cli-lab (0.0.10)
azure-cli-monitor (0.0.9)
azure-cli-network (2.0.13)
azure-cli-nspkg (3.0.1)
azure-cli-profile (2.0.11)
azure-cli-rdbms (0.0.6)
azure-cli-redis (0.2.8)
azure-cli-resource (2.0.13)
azure-cli-role (2.0.11)
azure-cli-servicefabric (0.0.3)
azure-cli-sf (1.0.8)
azure-cli-sql (2.0.10)
azure-cli-storage (2.0.14)
azure-cli-vm (2.0.13)
azure-common (1.1.8)
azure-datalake-store (0.0.15)
azure-graphrbac (0.31.0)
azure-keyvault (0.3.4)
azure-mgmt-authorization (0.30.0)
azure-mgmt-batch (4.1.0)
azure-mgmt-billing (0.1.0)
azure-mgmt-cdn (0.30.2)
azure-mgmt-cognitiveservices (1.0.0)
azure-mgmt-compute (2.0.0)
azure-mgmt-consumption (0.1.0)
azure-mgmt-containerinstance (0.1.0)
azure-mgmt-containerregistry (0.3.1)
azure-mgmt-containerservice (1.0.0)
azure-mgmt-datalake-analytics (0.1.6)
azure-mgmt-datalake-nspkg (2.0.0)
azure-mgmt-datalake-store (0.1.6)
azure-mgmt-devtestlabs (2.0.0)
azure-mgmt-dns (1.0.1)
azure-mgmt-documentdb (0.1.3)
azure-mgmt-iothub (0.2.2)
azure-mgmt-keyvault (0.40.0)
azure-mgmt-monitor (0.2.1)
azure-mgmt-network (1.4.0)
azure-mgmt-nspkg (2.0.0)
azure-mgmt-rdbms (0.1.0)
azure-mgmt-redis (4.1.0)
azure-mgmt-resource (1.1.0)
azure-mgmt-servicefabric (0.1.0)
azure-mgmt-sql (0.6.0)
azure-mgmt-storage (1.2.0)
azure-mgmt-trafficmanager (0.30.0)
azure-mgmt-web (0.32.0)
azure-monitor (0.3.0)
azure-multiapi-storage (0.1.3)
azure-nspkg (2.0.0)
azure-servicefabric (5.6.130)
bcrypt (3.1.3)
certifi (2017.7.27.1)
cffi (1.10.0)
chardet (3.0.4)
colorama (0.3.9)
cryptography (2.0.3)
futures (3.1.1)
gyp (0.1)
humanfriendly (4.4.1)
idna (2.6)
isodate (0.5.4)
jmespath (0.9.3)
keyring (10.4.0)
mock (2.0.0)
msrest (0.4.14)
msrestazure (0.4.13)
oauthlib (2.0.2)
paramiko (2.2.1)
pbr (3.1.1)
pip (9.0.1)
prompt-toolkit (1.0.15)
pyasn1 (0.3.3)
pycparser (2.18)
pydocumentdb (2.2.0)
Pygments (2.2.0)
PyJWT (1.5.2)
PyNaCl (1.1.2)
pyodbc (4.0.17)
pyOpenSSL (17.2.0)
python-dateutil (2.6.1)
pytz (2017.2)
PyYAML (3.12)
requests (2.18.4)
requests-oauthlib (0.8.0)
scp (0.10.2)
SecretStorage (2.3.1)
setuptools (18.0.1)
six (1.10.0)
sshtunnel (0.1.2)
tabulate (0.7.7)
tzlocal (1.4)
unittest-xml-reporting (2.1.0)
urllib3 (1.22)
virtualenv (13.1.2)
vsts-cd-manager (0.118.0)
wcwidth (0.1.7)
Whoosh (2.7.4)
xmltodict (0.11.0)
```

**Comment by troydai** [comment]
I can confirm this issue. A fix is being worked on.

**Comment by TwinkleChawla** [comment]
Hey, I'm also getting the same issue. And I'm having maximum of 3GB data in .tar.gz file. Please help!

**Comment by troydai** [comment]
It is fixed in 2.0.18 release. Please try it out.
